accounts/backend.py

```diff
@@ -0,0 +1,42 @@
+from django.conf import settings
+from social.backends.oauth import BaseOAuth2
+from urllib.parse import urlencode
+import json
+import urllib
+
+
+class OpenStaxOAuth2(BaseOAuth2):
+    """openstax OAuth authentication backend"""
+    name = 'openstax'
+    AUTHORIZATION_URL = settings.AUTHORIZATION_URL 
+    ACCESS_TOKEN_URL = settings.ACCESS_TOKEN_URL 
+    USER_QUERY = settings.USER_QUERY
+    REQUEST_TOKEN_METHOD = 'POST'
+    ACCESS_TOKEN_METHOD = 'POST'
+    SCOPE_SEPARATOR = ','
+    EXTRA_DATA = [
+        ('id', 'id'),
+    ]
+
+    def get_user_details(self, response):
+        """Return user details from openstax account's"""
+        return {'id': response.get('id'),
+                'username': response.get('username'),
+                'first_name':response.get('first_name'),
+                'last_name':response.get('last_name'),
+                'full_name': response.get('full_name')}
+
```

Comment:
Emails are stored in the database but they are not one of the fields passed back by the api e.g. 

``https://accounts-qa.openstax.org/api/user?`` passes back 

``` json
{
  "id": 16180,
  "username": "openstax_cms_tester",
  "first_name": "Richard",
  "last_name": "Hart"
}
```